Classes/common/CDTDatastore.m

```diff
@@ -150,6 +168,11 @@ - (NSUInteger)documentCount
 
 - (NSString *)name { return self.database.name; }
 
+- (id<CDTEncryptionKeyProvider>)copyEncryptionKeyProvider
+{
+    return [_encryptionKeyProvider copy];
+}
```

Comment:
Ok, in that case it is not worth to copy the key provider at all, even during the initialisation of a datastore. I'll remove the `NSCopying` from `CDTEncryptionKeyProvider` as well as the method `copy`.